### PR TITLE
Feature/add viper configfile

### PR DESCRIPTION
**Pull request by talhaw**
added viper/config file for configuration

### CODE REVIEW

File: configuration/config.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+const (
+	DbName = "db.name"
+)
+
+func init() {
+	viper.SetDefault("ContentDir", "content")
+	viper.SetDefault("LayoutDir", "layouts")
+	viper.SetDefault("Taxonomies", map[string]string{"tag": "tags", "category": "categories"})
```
Comment (ehmad-kamal-99) [inline configuration/config.go]: Do we need this in our project? I dont understand its usage.

File: configuration/config.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	DbName = "db.name"
+	DbHost="db.ip"
+	DbPort="db.port"
+	DbUser="db.user"
+	DbPass="db.pass"
+)
+
+func init() {
+	// default flags
+	pflag.String("coun", "", "sets db name")
```
Comment (ahab94) [inline configuration/config.go]: Lets remove flags for now...

File: configuration/config.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package configuration
+
+import (
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	DbName = "db.name"
+	DbHost="db.ip"
```
Comment (ahab94) [inline configuration/config.go]: make sure to run `make format`

File: models/teacher_test.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
+package models
```
Comment (ahab94) [inline models/teacher_test.go]: These are older changes please make sure these are not part of this pr.

File: scripts/test.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (ahab94) [inline scripts/test.sh]: Make sure to rebase.

File: db/mysql/client.go [hunk 6]
```diff
@@ -10,14 +10,15 @@ import (
...
 )
 
 const (
-	databaseName     = "apna_school"
+	databaseName     = configuration.DbName
 	studentTableName = "student"
 	teacherTableName = "teacher"
-	databasePassword = "Talha1996@gmail.com"
-	databaseUser     = "talha"
+	databasePassword = configuration.DbPass
```
Comment (ahab94) [inline db/mysql/client.go]: this should be `viper.GetString(configuration.DbPass)` same for other occassions.

remember `configration.DbPass` is just a key. To access it's value which we stored in the viper.. we do
`viper.GetString(configuration.DbPass)`

File: Makefile [hunk 7]
```diff
@@ -15,14 +16,14 @@ tmp/dev_image_id:
...
-.PHONY: check
 check: prepare
 	${DOCKRUN} bash /opt/check.sh
 
-.PHONY: todo
 todo: prepare
 	${DOCKRUN} bash /opt/todo.sh
+
+test: prepare format check
+	${DOCKRUN} bash /opt/test.sh
```
Comment (ahab94) [inline Makefile]: Please rebase and don't add this file in your changes.

File: scripts/test.sh (hunk 5 above)
Comment (talhaw) [reply scripts/test.sh]: Okkk doing it

File: configuration/config.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+const (
+	DbName = "db.name"
+	DbHost = "db.ip"
+	DbPort = "db.port"
+	DbUser = "db.user"
+	DbPass = "db.pass"
+)
+
+func init() {
+	viper.SetEnvPrefix("apna_school")
```
Comment (ahab94) [inline configuration/config.go]: we don't need env prefix. because will use containers for deploy.

File: db/mongo/client.go [hunk 9]
```diff
@@ -2,6 +2,8 @@ package mongo
 
 import (
 	"context"
+	"github.com/ahab94/ApnaSchool/configuration"
```
Comment (ahab94) [inline db/mongo/client.go]: make sure to arrange imports properly.

1: golang packages

2: 3rd party

3: project packages

File: db/mongo/client.go [hunk 10]
```diff
@@ -15,7 +17,7 @@ import (
 )
 
 const (
-	dbName        = "ApnaSchool"
+	dbName        = viper.GetString(configuration.DbName)
```
Comment (ahab94) [inline db/mongo/client.go]: we are not going to use `const` anymore. use `viper.GetString(configuration.DbName)` directly where it's needed.

File: db/mysql/client.go [hunk 11]
```diff
@@ -2,22 +2,24 @@ package mysql
 
 import (
 	"fmt"
+	"github.com/spf13/viper"
```
Comment (ahab94) [inline db/mysql/client.go]: arrange imports.

File: db/mysql/client.go [hunk 12]
```diff
@@ -2,22 +2,24 @@ package mysql
...
 	"github.com/pkg/errors"
 
+	"github.com/ahab94/ApnaSchool/configuration"
 	"github.com/ahab94/ApnaSchool/db"
 	"github.com/ahab94/ApnaSchool/models"
 )
 
 const (
-	databaseName     = "apna_school"
+	databaseName     = viper.GetString(configuration.DbName)
```
Comment (ahab94) [inline db/mysql/client.go]: We cannot instantiated consts like this. we are not going to use consts now that we have these values configurable. use viper directly where needed.

File: db/mysql/client.go [hunk 13]
```diff
@@ -32,7 +34,7 @@ type client struct {
 func formatDSN() string {
 	cfg := mysql.NewConfig()
 	cfg.Net = "tcp"
-	cfg.Addr = fmt.Sprintf("%s:%s", "localhost", "3306")
+	cfg.Addr = fmt.Sprintf("%s:%s", configuration.DbHost, configuration.DbPort)
```
Comment (ahab94) [inline db/mysql/client.go]: use viper.GetString(configuration.Dbhost) as i had mentioned before.
